src/camelot/fetchNitroPool.ts

```diff
@@ -0,0 +1,198 @@
+import { BigNumber, ethers } from 'ethers'
+
+import { CamelotNitroPool, ERC20 } from '../typechained'
+import { Geb } from '../geb'
+import { fromBigNumber, multicall, CamelotMulticallRequest, SECONDS_IN_YEAR } from '../utils'
+
+export type NitroPoolDetails = {
```

Comment:
Ensure that at minimum, all the necessary details are returned from this function. See https://github.com/open-dollar/od-app/blob/af447bf53530e8535a310ec26c9021adb8af53ad/src/containers/Earn/index.tsx

Reply:
I ensured to fetch back as much data as is available on the contract